torch/csrc/jit/codegen/cuda/compute_at_map.cpp

```diff
@@ -931,130 +869,233 @@ IterDomain* ComputeAtMap::computeConcreteId(
+  // backwards to inputs and remove every exact ID set from "traversed IDs".
+  //
+  // Remove (2) and (3) from (1) and we have the iteration domains we must
+  // resolve. The concrete ID must be in that set.
+  //
+  // Find any maybe concrete ID through the same iter/broadcast counting as
+  // before as it should work fine.
 
-  // For the LOOP map, the concrete ID must account for all root IDs
-  // of all of the IDs in each disjoit set. At least those ID's that are
-  // non-broadcast/non-reduction. As broadcast is only important here if it's
-  // concretized in the set. Track information so we can later make sure the
-  // concrete id has accounted for all iter domains meaning it has a correct
-  // loop size.
-  std::unordered_set<IterDomain*> root_ids_of_all_ids;
-  std::unordered_set<IterDomain*> root_ids_of_concrete_id;
-  std::unordered_map<IterDomain*, std::vector<IterDomain*>>
-      root_id_to_maybe_concrete_ids;
-
-  // Populate the above information, look for the concrete id, validate the loop
-  // concrete ID.
-  for (auto maybe_concrete_id : maybe_concrete_ids.vector()) {
-    std::unordered_set<IterDomain*> root_ids;
-    std::deque<IterDomain*> to_visit;
-
-    to_visit.push_back(maybe_concrete_id);
-    while (to_visit.size()) {
-      auto current_id = to_visit.front();
-      to_visit.pop_front();
-      if (isViewRfactor(current_id)) {
-        root_ids.emplace(current_id);
+  VectorOfUniqueEntries<std::shared_ptr<VectorOfUniqueEntries<IterDomain*>>>
+      maybe_concrete_exact_sets;
+
+  for (auto maybe_concrete_id : maybe_concrete_ids) {
+    maybe_concrete_exact_sets.pushBack(
+        disjointSetOf(maybe_concrete_id, IdMappingMode::EXACT));
+  }
+
+  // Going to iteratively modify this to be all sets that the concrete ID needs
+  // to cover
+  VectorOfUniqueEntries<std::shared_ptr<VectorOfUniqueEntries<IterDomain*>>>
+      all_exact_sets_covered =
+          getAllDisjointSetProducers(maybe_concrete_exact_sets);
+
+  // Remove all broadcast domains that are resolved within the history of any of
+  // the maybe concrete sets.
+  {
+    // All broadcast exact sets in all_exact_sets_covered that are resolved by
+    // IterDomains in all_exact_sets_covered
+    VectorOfUniqueEntries<std::shared_ptr<VectorOfUniqueEntries<IterDomain*>>>
+        resolved_broadcasts;
+
+    for (auto exact_set : all_exact_sets_covered) {
+      TORCH_INTERNAL_ASSERT(
+          exact_set->vector().size(),
+          "Cannot compute concrete id of empty set.");
+      auto c_id = getConcreteMappedID(
+          exact_set->vector().front(), IdMappingMode::EXACT);
+      if (!c_id->isBroadcast()) {
         continue;
       }
 
-      // push back producer IterDomains or add root if they don't exist
-      auto producer_vals = ir_utils::producerValsOf(current_id);
-      auto producer_ids = ir_utils::filterByType<IterDomain>(producer_vals);
+      if (trivial_reduction_info_.isDerived(c_id)) {
```

Comment:
Note to self: Can a broadcast domain be detected as a trivial reduction domain?

Reply:
I think in practice the trivial reduction wouldn't be the concrete id, but the concrete id would be an unresolved broadcast. This is out of extreme caution that the concrete ID could end up as the concrete ID.

I believe the reason a concrete id wouldn't be trivial reduction is because we just grab the first ID in the unique vector, which is generated from producer->consumers, so it's just highly unlikely the first dim in the set is a trivial reduction.

Reply:
Hmm, not sure if I understood your comment correctly. Here, `c_id` is a broadcast domain. I don't think we would mark any broadcast domain as a trivial reduction domain.

Reply:
Oh, this seems wrong then. What we're trying to do is remove any exact sets that the concrete ID doesn't need to "resolve". So any broadcast sets that have a resolution within the set we're trying to compute the concrete ID should be removed because there's the non-broadcast version of that ID somewhere.

What we want to do is also remove any set that's just broadcasts and a trivial reduction. So we should also catch the groups that are just broadcast->trivial reduction because that's not something we need resolved. That will get resolve through almost exact index propagation.

Reply:
This should just be a section in here that's if c_id is broadcast or trivial id, and if broadcast that broadcast is trivially reduced.